### PR TITLE
New generic rare golden emote modifier with a 1.5% proc chance thats deactivated during halloween

### PR DESCRIPTION
**Pull request by x-Xymos**
https://github.com/MemeLabs/chat-gui/issues/39
what it looks like https://codepen.io/Xymos/pen/JQyOWL

### CODE REVIEW

**Comment by Oreganon** [comment]
Way to common in my opinion. I think it should be truly rare with about one occurrence per day.

There were 277840 emotes used in september, thats about 9000 emotes per day. So I suggest  1 / 9000 = 0.00011111111 as the probability.

**Comment by x-Xymos** [comment]
> Way to common in my opinion. I think it should be truly rare with about one occurrence per day.
> 
> There were 277840 emotes used in september, thats about 9000 emotes per day. So I suggest 1 / 9000 = 0.00011111111 as the probability.

1 a day sounds kind of low, why not something like 10-20 a day, what does everyone else think.

**Comment by jbpratt78** [comment]
I like the idea of it being incredibly rare, even more so than a 'once a day' chance. Would love to hear others thoughts, maybe even poll a few members of the community.

**Comment by Oreganon** [comment]
Now that I think about it, once per day would get boring very soon I think. It would be too common so you would get used to it and it would lose it special appeal. Maybe even once per month would be nice.

**Comment by SoMuchForSubtlety** [comment]
Once a month still seems very high, how about once a decade?

**Comment by SmCTwelve** [comment]
Correlate the probability with how often Dashh appears in chat.

**Comment by x-Xymos** [comment]
It's turned off for halloween anyway so we have time to gather votes from chat before we decide to merge https://www.strawpoll.me/18764578/r

File: assets/chat/js/formatters.js [hunk 1]
```diff
@@ -178,26 +203,41 @@ class EmoteFormatter {
             if (message != null) {
                 timestamp = message.timestamp._i;
             }
-
+
```
Comment (jbpratt78) [inline assets/chat/js/formatters.js]: Is this extra white space or just GitHub?

File: assets/chat/js/formatters.js [hunk 2]
```diff
@@ -178,26 +203,41 @@ class EmoteFormatter {
...
-            if (isHalloween() && emoteCount <= 7 && proc(seed, punish)) {
+            if (isHalloween() && emoteCount <= 7 && proc(seed, punish,0)) {
                 innerClasses.push(getRandomHalloweenEffect(emote, seed));
             }
 
             if (chat.settings.get('animateforever')) {
                 innerClasses.push('chat-emote-'+emote+'-animate-forever')
             }
-
+
```
Comment (jbpratt78) [inline assets/chat/js/formatters.js]: Here as well

File: assets/chat/js/formatters.js [hunk 3]
```diff
@@ -178,26 +203,41 @@ class EmoteFormatter {
             if (message != null) {
                 timestamp = message.timestamp._i;
             }
-
+            
             const seed = genSeed(str, chat, i++, timestamp);
             // since the rng mostly depends on the two last messages, combos after stuck proc-ing a lot. Lower chance of this happening.
             const punish = str == getLastMsg(chat)
-            if (isHalloween() && emoteCount <= 7 && proc(seed, punish)) {
+            if (isHalloween() && emoteCount <= 7 && proc(seed, punish,0)) {
```
Comment (jbpratt78) [inline assets/chat/js/formatters.js]: add a space `..h, 0)`

**Pull request merged**